internal/services/tesla_fleet_api_service.go

```diff
@@ -91,50 +90,98 @@ func (t *teslaFleetAPIService) CompleteTeslaAuthCodeExchange(ctx context.Context
-// @Returns     {array} []services.TeslaVehicle
 func (t *teslaFleetAPIService) GetVehicles(ctx context.Context, token, region string) ([]TeslaVehicle, error) {
 	baseURL := fmt.Sprintf(t.Settings.Tesla.FleetAPI, region)
 	url := baseURL + "/api/1/vehicles"
 
+	resp, err := teslaHTTPClientHelper(ctx, t.log, t.HTTPClient, url, token)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch vehicles for user: %w", err)
+	}
+	defer resp.Body.Close()
+
+	vehicles := new(GetVehiclesResponse)
+	if err := json.NewDecoder(resp.Body).Decode(vehicles); err != nil {
+		return nil, fmt.Errorf("invalid response encountered while fetching user vehicles: %w", err)
+	}
+
+	if vehicles.Response == nil {
+		return nil, fmt.Errorf("error occurred fetching user vehicles")
+	}
+
+	return vehicles.Response, nil
+}
+
+// GetVehicle
+// @Description Call Tesla Fleet API to get a single vehicle by ID
+func (t *teslaFleetAPIService) GetVehicle(ctx context.Context, token, region string, vehicleID int) (*TeslaVehicle, error) {
+	baseURL := fmt.Sprintf(t.Settings.Tesla.FleetAPI, region)
+	url := fmt.Sprintf("%s/api/1/vehicles/%d", baseURL, vehicleID)
+
+	resp, err := teslaHTTPClientHelper(ctx, t.log, t.HTTPClient, url, token)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch vehicles for user: %w", err)
+	}
+	defer resp.Body.Close()
+
+	vehicle := new(struct {
+		Response TeslaVehicle `json:"response"`
+	})
+	if err := json.NewDecoder(resp.Body).Decode(vehicle); err != nil {
+		return nil, fmt.Errorf("invalid response encountered while fetching user vehicles: %w", err)
+	}
+
+	return &vehicle.Response, nil
+}
+
+// WakeUpVehicle
+// @Description Call Tesla Fleet API to wake a vehicle from sleep
+func (t *teslaFleetAPIService) WakeUpVehicle(ctx context.Context, token, region string, vehicleID int) error {
+	baseURL := fmt.Sprintf(t.Settings.Tesla.FleetAPI, region)
+	url := fmt.Sprintf("%s/api/1/vehicles/%d/wake_up", baseURL, vehicleID)
+
+	resp, err := teslaHTTPClientHelper(ctx, t.log, t.HTTPClient, url, token)
+	if err != nil {
+		return fmt.Errorf("could not fetch vehicles for user: %w", err)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("got status code %d waking up vehicle %d", resp.StatusCode, vehicleID)
+	}
+
+	return nil
+}
+
+func teslaHTTPClientHelper(ctx context.Context, logger *zerolog.Logger, httpClient *http.Client, url, token string) (*http.Response, error) {
 	ctxTimeout, cancel := context.WithTimeout(ctx, time.Second*10)
 	defer cancel()
 
 	req, err := http.NewRequestWithContext(ctxTimeout, "GET", url, nil)
 	if err != nil {
-		return nil, fmt.Errorf("could not fetch vehicles for user: %w", err)
+		return nil, err
 	}
 	req.Header.Set("Authorization", "Bearer "+token)
 
-	resp, err := t.HTTPClient.Do(req)
+	resp, err := httpClient.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("could not fetch vehicles for user: %w", err)
+		return nil, err
```

Comment:
I think providing the context in these errors with `fmt.Errorf` was good, we should keep doing it.